src/parsing/prscParser.ts

```diff
@@ -363,16 +368,49 @@ function generateParser(options: { outputDebugInfo: boolean; xquery: boolean }):
 
 	const nameTest: Parser<IAST> = or([wildcard, map(eqName, (x) => ['nameTest', ...x])]);
 
-	const nodeTest: Parser<IAST> = or([kindTest, nameTest]);
+	const simpleNodeTest: Parser<IAST> = or([kindTest, nameTest]);
 
-	const abbrevForwardStep: Parser<IAST> = then(optional(tokens.AT_SIGN), nodeTest, (a, b) => {
-		return a !== null || isAttributeTest(b)
-			? ['stepExpr', ['xpathAxis', 'attribute'], b]
-			: ['stepExpr', ['xpathAxis', 'child'], b];
-	});
+	const unionNodeTest: Parser<IAST[]> = delimited(
+		followed(tokens.BRACE_OPEN, whitespace),
+		cut(
+			then(
+				simpleNodeTest,
+				star(preceded(surrounded(tokens.VERTICAL_BAR, whitespace), simpleNodeTest)),
+				(first, rest) => [first].concat(rest),
+			),
+		),
+		followed(whitespace, tokens.BRACE_CLOSE),
+	);
+
+	const nodeTest: Parser<IAST[]> =
+		options.version === 4
+			? or([unionNodeTest, map(simpleNodeTest, (test) => [test])])
+			: map(or([kindTest, nameTest]), (test) => [test]);
+
+	const abbrevForwardStep: Parser<IAST> =
+		options.version === 4
+			? or([
+					map(
+						preceded(tokens.AT_SIGN, nodeTest),
+						(test) => ['stepExpr', ['xpathAxis', 'attribute'], test] as IAST,
+					),
+					map(
+						simpleNodeTest,
+						(test) => ['stepExpr', ['xpathAxis', 'child'], test] as IAST,
+					),
+			  ])
+			: then(optional(tokens.AT_SIGN), simpleNodeTest, (a, b) => {
+					return a !== null || isAttributeTest(b)
+						? ['stepExpr', ['xpathAxis', 'attribute'], b]
+						: ['stepExpr', ['xpathAxis', 'child'], b];
+			  });
 
 	const forwardStep: Parser<IAST> = or([
-		then(forwardAxis, nodeTest, (axis, test) => ['stepExpr', ['xpathAxis', axis], test]),
+		then(
+			forwardAxis,
+			nodeTest,
+			(axis, tests) => (['stepExpr', ['xpathAxis', axis]] as IAST).concat(tests) as IAST,
```

Comment:
Is there an XQueryX spec for 4.0?

Reply:
Nope not yet. Brought it up with the community group already